packages/components/src/date-time/time.js

```diff
@@ -120,46 +180,127 @@ class TimePicker extends Component {
 		this.setState( { minutes: event.target.value } );
 	}
 
+	maxHours() {
+		return ( this.props.is12Hour ? 12 : 23 );
+	}
+
+	minHours() {
+		return ( this.props.is12Hour ? 1 : 0 );
+	}
+
 	render() {
 		const { is12Hour } = this.props;
-		const { minutes, hours, am } = this.state;
+		const { day, month, year, minutes, hours, am } = this.state;
 
 		return (
-			<div className="components-time-picker">
-				<input
-					className="components-time-picker__input"
-					type="text"
-					value={ hours }
-					onChange={ this.onChangeHours }
-					onBlur={ this.updateHours }
-				/>
-				<span className="components-time-picker__separator">:</span>
-				<input
-					className="components-time-picker__input"
-					type="text"
-					value={ minutes }
-					onChange={ this.onChangeMinutes }
-					onBlur={ this.updateMinutes }
-				/>
-				{ is12Hour && <div>
-					<Button
-						isDefault
-						className="components-time-picker__am-button"
-						isToggled={ am === 'AM' }
-						onClick={ this.updateAmPm( 'AM' ) }
-					>
-						{ __( 'AM' ) }
-					</Button>
-					<Button
-						isDefault
-						className="components-time-picker__pm-button"
-						isToggled={ am === 'PM' }
-						onClick={ this.updateAmPm( 'PM' ) }
-					>
-						{ __( 'PM' ) }
-					</Button>
-				</div> }
-			</div>
+			<fieldset
+				className={
+					is12Hour ? 'components-datetime__time components-datetime__time--12-hour' : 'components-datetime__time components-datetime__time--24-hour'
+				}
+			>
+				<legend>{ __( 'Publish at' ) }</legend>
+				<fieldset className="invisible">
+					<legend className="invisible">{ __( 'Date' ) }</legend>
+					<div className="components-datetime__time__form">
+						<div className="components-datetime__time__form__container components-datetime__time__form__container--month">
+							<select
+								aria-label={ __( 'Month' ) }
```

Comment:
Is it better to use an `aria-label` and no actual `<label>` here? It seems off to me.

cc @afercia

Reply:
Works the same. This was actually done after @afercia's initial review.